packages/monaco/src/browser/monaco-workspace.ts

```diff
@@ -162,7 +166,7 @@ export class MonacoWorkspace {
         if (this.suppressedOpenIfDirty.indexOf(model) !== -1) {
             return;
         }
-        if (model.dirty && MonacoEditor.findByDocument(this.editorManager, model).length === 0) {
+        if (model.dirty && MonacoEditor.findByDocument(this.editorManager, model).length === 0 && !CustomEditorWidget.is(this.shell.activeWidget)) {
```

Comment:
I believe the `monaco-workspace` should only deal with underlying `models` and should not be made aware or deal with widgets (especially the `shell`).

Reply:
Removed shell and CustomeEditorWidget. Please see if solution is acceptable.